qutebrowser/utils/urlutils.py

```diff
@@ -682,3 +682,25 @@ def widened_hostnames(hostname: str) -> Iterable[str]:
     while hostname:
         yield hostname
         hostname = hostname.partition(".")[-1]
+
+
+def get_url_yank_text(url: QUrl, is_pretty: bool) -> str:
```

Comment:
IMHO, using

```suggestion
def get_url_yank_text(url: QUrl, *, is_pretty: bool) -> str:
```

would be good to force calling code to not use a magic boolean there. Currently, with e.g.:

```python
urlutils.get_url_yank_text(QUrl(question.url), False)
```

it's completely opaque what this magic `False` means. In C/C++ people often use enums or comments to fix this, in Python usually keyword-arguments are used, i.e. `pretty=False`.


I'd also prefer:

```suggestion
def get_url_yank_text(url: QUrl, *, pretty: bool) -> str:
```

because `is_pretty` sounds to me like we're telling the function whether the URL is pretty.

Reply:
I agree, this is better. Thanks! I pushed a commit to implement this.